File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        letters_count = {}

        for l_s in s:
            if(l_s in letters_count):
                letters_count[l_s] += 1
            else:
                letters_count[l_s] = 1

        for t_s in t:
            if(t_s in letters_count):
                letters_count[t_s] -= 1
            else:
                return False

        return all(value == 0 for value in letters_count.values())

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        const len = nums.length
        const res = new Array(len).fill(1);

        // Calculating all left multiplications
        let left = nums[0];
        for(let i = 1; i < len; i++) {
            res[i] *= left;
            left *= nums[i];
        }

        console.log(res)

        let right = nums[len - 1];
        for(let i = len - 2; i >= 0; i--) {
            res[i] *= right;
            right *= nums[i];
        }

        return res;
    }
}
